src/post-preview.js

```
@@ -14,15 +17,35 @@ function PostPreview (options) {
     instance.element = document.querySelector(options.element);
   }
 
+  var previewClient = new SolidusClient();
+
   options = _setDefaults(options, instance);
 
   if (options.resource) {
-    _getJSON(options.resource, function(resp){
-        options.content = resp;
-        //Render the Default Template
-        instance.element.outerHTML = instance.template(options);
-        //Select new DOM element after replacing original with rendered template
+
+    var preprocessor = function(context) {
+      // Render the preview resource directly, allows re-use of server-side templates
+      context = context.resources.preview;
+      return context;
+    };
+
+    var view = {
+      resources: {
+        preview: {
+          url: options.resource,
+          with_credentials: true,
+          query: {'_wp_json_nonce': params._wp_json_nonce}
+        }
+      },
+      preprocessor: preprocessor,
+      template: instance.template
+    };
+
+    previewClient.render(view).end(function(err, html) {
```

Should throw the error if there is one here